avm/ptn/deployment-script/import-image-to-acr/tests/e2e/max/dependencies.bicep

```
@@ -0,0 +1,165 @@
+    properties: {
+      addressPrefix: cidrSubnet(ipRange, 24, 1)
+      serviceEndpoints: [
+        {
+          service: 'Microsoft.Storage'
+        }
+      ]
+      delegations: [
+        {
+          name: 'Microsoft.ContainerInstance.containerGroups'
+          properties: {
+            serviceName: 'Microsoft.ContainerInstance/containerGroups'
+          }
+        }
+      ]
+    }
+  }
+}
+
+module dnsZoneContainerRegistry 'br/public:avm/res/network/private-dns-zone:0.2.5' = {
+  name: '${uniqueString(deployment().name, location)}-dnsZone-ACR'
+  params: {
+    name: 'privatelink.azurecr.io'
+    virtualNetworkLinks: [
+      {
+        name: '${vnet.name}-ContainerRegistry-link'
+        virtualNetworkResourceId: vnet.id
+        registrationEnabled: false
+      }
+    ]
+  }
+}
+
+module privateEndpointContainerRegistry 'br/public:avm/res/network/private-endpoint:0.4.1' = {
+  name: '${uniqueString(deployment().name, location)}-pe-ACR'
+  params: {
+    name: '${uniqueString(resourceGroup().name, location)}-pe-ContainerRegistry'
+    subnetResourceId: vnet::subnet_privateendpoints.id
+    customNetworkInterfaceName: '${uniqueString(resourceGroup().name, location)}-pe-ContainerRegistry-nic'
+    location: location
+    privateDnsZoneGroupName: 'default'
+    privateDnsZoneResourceIds: [
+      dnsZoneContainerRegistry.outputs.resourceId
+    ]
+    privateLinkServiceConnections: [
+      {
+        name: 'pe-ContainerRegistry-connection'
+        properties: {
+          privateLinkServiceId: acr.outputs.resourceId
+          groupIds: [
+            'registry'
+          ]
+        }
+      }
+    ]
+  }
+}
+
+module storage 'br/public:avm/res/storage/storage-account:0.9.0' = {
+  name: '${uniqueString(resourceGroup().name, location)}-storage'
+  params: {
+    name: storageAccountName
+    location: location
+    kind: 'StorageV2'
+    minimumTlsVersion: 'TLS1_2'
+    skuName: 'Standard_LRS'
+    accessTier: 'Hot'
+    allowSharedKeyAccess: true
+    allowBlobPublicAccess: false
+    networkAcls: {
+      bypass: 'AzureServices'
+      virtualNetworkRules: [
+        {
+          id: vnet::subnet_deploymentscript.id
+          action: 'Allow'
+        }
+      ]
+      defaultAction: 'Deny'
+    }
+    roleAssignments: [
+      {
+        principalId: identity.outputs.principalId
+        principalType: 'ServicePrincipal'
+        roleDefinitionIdOrName: '69566ab7-960f-475b-8e7c-b3118f30c6bd' // Storage File Data Privileged Contributor
+      }
+    ]
+  }
+}
+
+// the container registry to upload the image into
+module acr 'br/public:avm/res/container-registry/registry:0.2.0' = {
+  name: '${uniqueString(resourceGroup().name, location)}-acr'
+  params: {
+    name: acrName
+    location: location
+    acrSku: 'Premium'
+    acrAdminUserEnabled: false
+    roleAssignments: [
+      // assign ArcPull and AcrPush
+      for registryRole in ['7f951dda-4ed3-4680-a7ca-43fe172d538d', '8311e382-0749-4cb8-b61a-304f252e45ec']: {
```

```suggestion
      for registryRole in ['AcrPull', 'AcrPush']: {
```

The ACR module has the roles actually implemented :) The same can unfortunately not be said about the storage account module where the 'Storage File Data Privileged Contributor' is missing (and should be added)

---

This is like using a comment as code. Love it :-D

---

The future is neigh